include/envoy/request_info/request_info.h

```diff
@@ -68,66 +68,125 @@ class RequestInfo {
    */
   virtual void onUpstreamHostSelected(Upstream::HostDescriptionConstSharedPtr host) PURE;
 
+  /**
+   * @return the # of body bytes received in the request.
+   */
+  virtual uint64_t bytesReceived() const PURE;
+
+  /**
+   * @return the protocol of the request.
+   */
+  virtual const Optional<Http::Protocol>& protocol() const PURE;
+
+  /**
+   * Set the request's protocol.
+   */
+  virtual void protocol(Http::Protocol protocol) PURE;
+
+  /**
+   * @return the response code.
+   */
+  virtual const Optional<uint32_t>& responseCode() const PURE;
+
   /**
    * @return the time that the first byte of the request was received.
    */
   virtual SystemTime startTime() const PURE;
 
   /**
-   * @return duration from request start to when the entire request was received from the
-   * downstream client in microseconds.
+   * @return the monotonic time that the first byte of the request was received. Duration
+  calculations should be made relative to this value.
+  */
+  virtual MonotonicTime startTimeMonotonic() const PURE;
+
+  /**
+   * @return the monotonic clock time when the last byte of the request was received.
```

Comment:
I see a bunch of comments about monotonic clock time with nanoseconds args.  Stale comments?

Reply:
Yep, will update